Sources/Compass.swift

```diff
@@ -42,10 +42,10 @@ public struct Compass {
     return nil
   }
 
-  static func parseAsURL(url: NSURL, payload: Any? = nil) -> Location? {
+  static func parseAsURL(_ url: URL, payload: Any? = nil) -> Location? {
```

Comment:
I think that `parseAsURL` is actually dealing with url components, so I rename it to `parseComponents`